docs/RFCS/data_protection/ssec.md

```diff
@@ -0,0 +1,104 @@
+
+
+### Requirements
+
+LeoStorage encrypts and decrypts objects because LeoStorage's CPU usage tends to be considerably lower than LeoGateway. Furthermore, in many cases, the number of LeoStorage's in a LeoFS storage system is larger than the number of LeoGateway and thus, load balancing can be expected.
+
+In each of the following operations, LeoStorage handles encryption and decryption of an object after an operation is requested from LeoGateway.
+
+* GET
+* Range GET
+* HEAD
+* PUT
+* Multipart Upload
+* COPY
+
+LeoGateway never caches objects to securely protect.
+
+
+### Request Handling on LeoGateway
+
+When LeoGateway receives an SSE-C request from a LeoFS' client, it retrieves the encryption key information from the request header, then it sends the information and the object *(only if PUT or Multipart Upload operation was requested)* to LeoStorage.
+
+When a requested object corresponds to Large Size Object, LeoGateway divides the object to plural chunks of it, then it sends each chunk with the encryption key information to LeoStorage in order to individually execute object encryption.
+
+
+#### Encryption key information in HTTP Header:
+
+| Name  | Description  |
+|----|----|
+|`x-amz-server-side​-encryption​-customer-algorithm`| Encryption algorithm *(AES 256)* |
+|`x-amz-server-side​-encryption​-customer-key` | 256-bit, base64-encoded encryption key |
+| `x-amz-server-side​-encryption​-customer-key-MD5` | Base64-encoded 128-bit MD5 digest of the encryption key |
+
+When LeoGateway accepts COPY operation, a client sends a request header which includes the following items. The destination object is encrypted by using those items.
+
+| Name  | Description  |
+|----|----|
+|`x-amz-copy-source​-server-side​-encryption​-customer-algorithm`| Encryption algorithm *(AES 256)* |
+|`x-amz-copy-source​-server-side​-encryption​-customer-key` | 256-bit, base64-encoded encryption key |
+| `x-amz-copy-source-​server-side​-encryption​-customer-key-MD5` | Base64-encoded 128-bit MD5 digest of the encryption key |
+
+
+See more details on [AWS S3-API - Protecting Data Using SSE-C](https://docs.aws.amazon.com/AmazonS3/latest/dev/ServerSideEncryptionCustomerKeys.html).
+
+### Encryption and decryption of an object on LeoStorage
+
+LeoStorage receives an object and an encryption key information when LeoGateway requests **WRITE operation**, then it creates an initialization vector *(IV)* which is commonly called a nonce *(number used once)*, then it encrypts the object based on **the encryption algorithm** and **the encryption key** and **the initialization vector**, finally it replicates and stores the encrypted object and metadata which includes the following items:
+
+* 256-bit, base64-encoded encryption key
```

Comment:
To be precise, it's `256-bit, key encryption key`
- Not base64-encoded
- Not encryption key itself